lib/ObjWriter/objwriter.cpp

```diff
@@ -550,22 +547,119 @@ extern "C" void EmitDebugFileInfo(ObjectWriter *OW, int FileId,
   OST.EmitCVFileDirective(FileId, FileName);
 }
 
-// This should be invoked at the end of function code emission to flush
-// debug function info.
-extern "C" void EmitDebugFunctionInfo(ObjectWriter *OW,
-                                      const char *FunctionName,
-                                      int FunctionSize) {
+static void EmitPDBDebugVarInfo(MCObjectStreamer &OST, const MCSymbol *Fn,
+                                DebugVarInfo LocInfos[], int NumVarInfos) {
+  MCContext &OutContext = OST.getContext();
+  const MCObjectFileInfo *MOFI = OutContext.getObjectFileInfo();
+  assert(MOFI->getObjectFileType() == MOFI->IsCOFF);
+
+  for (int i = 0; i < NumVarInfos; i++) {
+    // Emit an S_LOCAL record
+    DebugVarInfo var = LocInfos[i];
+    LocalSym sym;
+
+    memset(&sym, 0, sizeof(sym));
+
+    int sizeofSym = sizeof(LocalSym);
+    short rectyp = SymbolRecordKind::S_LOCAL;
+    short reclen = sizeof(rectyp) + sizeofSym + var.name.length() + 1;
```

Comment:
Well. We now have struct layout so we can compute the length of symbol upfront like this. Another way is to just emit two labels (start/end) and emit difference so that we don't make any mistake. But this is fine with me, too.
But should we use "SymRecord" struct for this? I'm seeing this pattern in other places that emits length and kind. We might add a function that takes this common pattern.